### PR TITLE
modularizes data and allows dynamic buttons

### PR DESCRIPTION
**Pull request by shawnpyates**
1. Moves hardcoded data into separate `data` folder to keep functional code small. 

2. Enables list items to render buttons dynamically.

### CODE REVIEW

**Comment by paracycle** [comment]
Shawn, this is good, but IMO is not enough encapsulation/abstraction. Here is why:

We will probably need to load raw data from a JSON/YAML file do construct these elements, instead of defining them as JS objects. For example, the location of a given Merchant and the time of the transaction will be defined in the data, and both the static map image and the `"Show Map"` URL will be generated from the location and the `subtitle` field will be generated using the relative offset from the transaction time.

If the encapsulation is done this way, it will harder to construct these objects from data.

In short, we need builder objects (given raw data) instead of data objects. Do you mind refactoring this with that in mind?
